micro_mg3_0.F90

```diff
@@ -4178,4 +4378,75 @@ pure subroutine micro_mg_get_cols(ncol, nlev, top_lev, mgncol, mgcols, &
 
 end subroutine micro_mg_get_cols
 
+! =======================================================================
+! time - implicit monotonic scheme
+! developed by sj lin, 2016
+! =======================================================================
+
+subroutine implicit_fall (dt, ktop, kbot, ze, vt, dp, q, precip, m1)
+    
+    implicit none
+    
+    integer, intent (in) :: ktop, kbot
+    
+    real(r8), intent (in) :: dt
+    
+    real(r8), intent (in), dimension (ktop:kbot + 1) :: ze
+    
+    real(r8), intent (in), dimension (ktop:kbot) :: vt, dp
+    
+    real(r8), intent (inout), dimension (ktop:kbot) :: q
+    
+    real(r8), intent (out), dimension (ktop:kbot) :: m1
+    
+    real(r8), intent (out) :: precip
+    
+    real(r8), dimension (ktop:kbot) :: dz, qm, dd
+    
+    integer :: k
```

Comment:
It would be really nice if the lines above were single-spaced (no extra space) and had a comment for each input parameter to document what is going on in this subroutine.

Reply:
Done